--- Makefile.bak	2012-03-13 17:34:39.000000000 -0600
+++ Makefile	2012-03-13 17:38:39.000000000 -0600
@@ -64,7 +64,7 @@
 	statics_loader.o\
 	toydispatch.o
 
-all: $(LIBOBJC).a $(LIBOBJCXX).so.$(VERSION)
+all: $(LIBOBJCXX).so.$(VERSION)
 
 $(LIBOBJCXX).so.$(VERSION): $(LIBOBJC).so.$(VERSION) $(OBJCXX_OBJECTS)
 	$(SILENT)echo Linking shared Objective-C++ runtime library...
@@ -99,12 +99,10 @@
 	$(SILENT)install -d $(LIB_DIR)
 	$(SILENT)install -m 444 $(LIBOBJC).so.$(VERSION) $(LIB_DIR)
 	$(SILENT)install -m 444 $(LIBOBJCXX).so.$(VERSION) $(LIB_DIR)
-	$(SILENT)install -m 444 $(LIBOBJC).a $(LIB_DIR)
 	$(SILENT)if [ "$(strip)" = "yes" ]; then \
 		echo Stripping installed libraries...; \
 		strip $(LIB_DIR)/$(LIBOBJC).so.$(VERSION); \
 		strip $(LIB_DIR)/$(LIBOBJCXX).so.$(VERSION); \
-		strip $(LIB_DIR)/$(LIBOBJC).a; \
 	fi
 	$(SILENT)echo Creating symbolic links...
 	$(SILENT)ln -sf $(LIB_DIR)/$(LIBOBJC).so.$(VERSION) $(LIB_DIR)/$(LIBOBJC).so
@@ -123,4 +121,3 @@
 	$(SILENT)rm -f $(OBJCXX_OBJECTS)
 	$(SILENT)rm -f $(LIBOBJC).so.$(VERSION)
 	$(SILENT)rm -f $(LIBOBJCXX).so.$(VERSION)
-	$(SILENT)rm -f $(LIBOBJC).a
